### ISSUE TITLE
Slug test data?

### ISSUE THREAD

**Issue by kaijagahm** [MFEh2o/db#65]
Where are the slug test data? New table? Check with Jake Zwart.

**Comment by kaijagahm** [comment]
Emailed Jake 9/2

**Comment by joneslabND** [comment]
Yes, I would expect similar data for CR, HB, and MO. Not sure if Jake has this or if it is on onedrive...

**Comment by kaijagahm** [comment]
Jake sent me some more data from 2012 and 2014. Still working on sorting through what it all means. Meeting with Jake soon to discuss this. In particular, need help understanding how to do the Ksat calculations, whether to include hydraulic head data and how, and what the date is for the undated 2012 samples.

**Comment by kaijagahm** [comment]
We've opted to include only the calculated Ksat measurements in the new PIEZOMETER_SLUG_TESTS table (or maybe PIEZOMETER_CONDUCTIVITY? The raw time/depth measurements will be put into some sort of file structure and stored in Box but not put into a table. SampleID's will be made. 

I had started making the sampleID's using the actual measurements as depthBottom and depthTop, but after today's discussion I think that's not right. Should probably leave depthBottom and depthTop as NA's.

**Comment by kaijagahm** [comment]
I think the next step here is to look through the files sent by Stephen and see if I can work out how to do the ksat calculations.

**Comment by kaijagahm** [comment]
Looked through the Maidment chapter, attached here: 
[Maidment1992_GroundwaterFlow.pdf](https://github.com/MFEh2o/db/files/5765559/Maidment1992_GroundwaterFlow.pdf)

The equation in question is:
![Screen Shot 2021-01-04 at 10 53 30](https://user-images.githubusercontent.com/37053323/103553222-25612700-4e7b-11eb-992a-41760663bf01.png)

And a corresponding diagram is:
![Screen Shot 2021-01-04 at 10 53 35](https://user-images.githubusercontent.com/37053323/103553185-1aa69200-4e7b-11eb-920f-78c8dca0e12e.png)

I wrote Steve to confirm my understanding of the equation, asking him what h was. Of course, immediately after sending the email, I figured out what h was. Ugh. I feel silly. But anyway--h is the height of the water in the piezometer at any given time point (i.e. it falls between H0 and H).

T0 is calculated from H, h, and H0. You compute the normalized unrecovered head difference (H-h)/(H-H0) at each time point. Then you run a linear regression on those normalized unrecovered head differences over time. T0 is the time value such that the regression line predicts a value of 0.37 for the normalized unrecovered head difference.

I think I know enough now to write a script to calculate ksat from measurements.

**Comment by kaijagahm** [comment]
Waiting to hear back from Steve to confirm we can use the same ksat equation

**Comment by kaijagahm** [comment]
I passed the email thread to @joneslabND and @ctsolomon on 7/20.

To summarize: I received assorted slug test data (in various formats) via email from Jake Zwart. I put it [here](https://app.box.com/folder/122131119317). The data needs to be cleaned, and ksat measurements need to be calculated. I started a script to try my hand at creating a ksat calculation function. The function isn't done, because we're waiting for Steve to respond about whether or not we can use the equation outlined in Maidment, or whether we have to do something different.

The script in question is "gh065_kSatScript.R", and it's in my working folder under "r_code/".

At the bottom of the script, I've outlined a path forward with this workflow. Reproducing it here:
Next steps once this function is created:
1. Add the `computeKsat` function to the MFEUtilities package. Can make a new .R file, e.g. "calculateKSat.R". Use the roxygen header tags and list out the parameters, either by copying the header from an existing function or by following package dev instructions in the resources linked in the MFEUtilities README document.
2. Be sure to run devtools::document() to document the function after you're done adding it. Then devtools::check(), and once checks are passed (no errors), commit and push to github. Now the function will be part of the package.
3. Create a *new script* (don't use this one!). Load in the data from where it's stored, here: https://app.box.com/folder/122131119317. Apply `computeKsat` to the data after doing whatever data cleaning is necessary.
4. With the ksat computations done, you can make a new database table and add this data to the database. Then keep the function in the MFEUtilities package for use with any future data.
